src/js_realm_object.hpp

```diff
@@ -415,6 +417,24 @@ void RealmObjectClass<T>::remove_all_listeners(ContextType ctx, ObjectType this_
     realm_object->m_notification_tokens.clear();
 }
 
+template<typename T>
+void RealmObjectClass<T>::get_property_type(ContextType ctx, ObjectType this_object, Arguments &args, ReturnValue &return_value) {
+    args.validate_maximum(1);
+
+    std::string property_name = Value::validated_to_string(ctx, args[0], "propertyName");
+    
+    auto realm_object = get_internal<T, RealmObjectClass<T>>(ctx, this_object);
+    if (!realm_object) {
+        throw std::runtime_error("Invalid 'this' object");
+    }
+
+    const Property* prop = realm_object->get_object_schema().property_for_public_name(property_name);
+    if (!prop) {
+        throw std::invalid_argument(util::format("No such property: %1", property_name));
+    }
+
+    return_value.set(prop->type_string());
```

Comment:
To get the JavaScript names, you need to call `TypeErrorException::type_string(prop)` (https://github.com/realm/realm-js/blob/master/src/js_realm.cpp#L78).

If `prop->type_string() == "mixed"`, you need to unwrap the property. It is something like:

```
auto obj = realm_object->obj();
DataType typ = obj.get().get_type();
// convert typ to string
```